src/generator/qbe.rs

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright 2021 Alexey Yerin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+use super::Generator;
+use crate::ast::types::Type;
+use crate::ast::*;
+use std::collections::HashMap;
+
+pub struct QbeGenerator {
+    tmp_counter: u32,
+    // Variable -> temporary mapping
+    scope: HashMap<String, String>,
+    buf: String,
+}
+
+impl Generator for QbeGenerator {
+    fn generate(prog: Module) -> String {
+        let mut generator = QbeGenerator {
+            tmp_counter: 0,
+            scope: HashMap::new(),
+            buf: String::new(),
+        };
+
+        for func in prog.func {
+            generator.generate_function(&func);
+        }
+
+        generator.buf
+    }
+}
+
+impl QbeGenerator {
+    /// Generates a function definition
+    fn generate_function(&mut self, func: &Function) {
+        // Clean up variable scope used by previous function
+        self.scope.clear();
+
+        // XXX: should *all* functions be exported?
```

Comment:
Agree, but I think `pub` would suit better since it tells what effect this is going to have. Also, `export` is frequently confused with "export as C ABI", which is the default in Antimony